### DISCUSSION TITLE
New DCMotor functions

### DISCUSSION THREAD

**Discussion by Gcopper1984** [pybricks/support#538] [Q&A]
Hi to everyone,

   I see that you changed something in DCMotor command. What's the difference between the old one and the new one? Are the new one compatible with the old?

Thank you.

**Comment by laurensvalk** [comment]
Nicely spotted! We have an open task to document this in #536.

This will be done this week or the next we're finalizing the 3.1 version.

Please keep testing the beta and let us know if you find more problems! Thanks! 😄

**Comment by Gcopper1984** [comment]
😄😄😄

I will test it for sure! I am very interested in this case because now I use a custom battery that I found on the great philohome:

https://www.philohome.com/pupbat/pupbat.htm

where I setup a boost dc dc converter to power the Technic Hub at a 10V or 11V to get more power (WARNING: this could break the brick). But I also tried that in this situation the read voltage function from the HUB return value of 9.81 V instead of the real value.

I hope that this new way to calculate pwm do not cut power for my model.

**Reply by laurensvalk** [reply]
We are now limiting the voltage to the motor to 9V just to be safe. For the Technic Small Motor, we are reducing it to 6V.

So if you use a battery with a higher voltage than that, PWM will automatically be reduced such that the motor output is still limited to 9V.

**Comment by ZPhilo** [comment]
Glad you liked it ;) 
To be safe, don't exceed 10V as this is the absolute maximum rating of motor driver IC inside hub ;) - Though this guy is really tough if real world conditions!

**Comment by Gcopper1984** [comment]
Hi @ZPhilo !

   Yes, I really appreciate your documentation about lego technic, it is full of information!

I am aware of the risks of using such a high voltage (I dissasembled the brick to check witch motor driver lego used and I know that I exceed its maximun voltage) but  like to live the risky life 😄😄😄


For me it is not good to limit the output voltage to some value and I hope I can use 100% of the input power. You can not use the NiZn battery anymore to get that some extra speed for your models. I also believe that changing the battery voltage is such an uncommon thing that it shouldn't be a problem for most people, I am aware that I can break my technic hub module
